src/dex/algebra/algebra-pool.ts

```diff
@@ -227,6 +228,123 @@ export class AlgebraEventPool extends StatefulEventSubscriber<PoolState> {
+    const balancesAndGlobalStateCalldata: MultiCallParams<
+      bigint | DecodedStateMultiCallResultWithRelativeBitmaps
+    >[] = [
+      {
+        target: this.token0,
+        callData: this.erc20Interface.encodeFunctionData('balanceOf', [
+          this.poolAddress,
+        ]),
+        decodeFunction: uint256ToBigInt,
+      },
+      {
+        target: this.token1,
+        callData: this.erc20Interface.encodeFunctionData('balanceOf', [
+          this.poolAddress,
+        ]),
+        decodeFunction: uint256ToBigInt,
+      },
+      {
+        target: this.stateMultiContract.options.address,
+        callData: this.stateMultiContract.methods
+          .getFullStateWithoutTicks(
+            this.factoryAddress,
+            this.token0,
+            this.token1,
+            0,
+            0,
+          )
+          .encodeABI(),
+        decodeFunction: decodeStateMultiCallResultWithRelativeBitmaps,
+      },
+    ];
+
+    const [resBalance0, resBalance1, stateWithoutTicksAndTickBitmap] =
+      await this.dexHelper.multiWrapper.tryAggregate<
+        bigint | DecodedStateMultiCallResultWithRelativeBitmaps
+      >(
+        false,
+        balancesAndGlobalStateCalldata,
+        blockNumber,
+        this.dexHelper.multiWrapper.defaultBatchSize,
+        false,
+      );
+
+    assert(stateWithoutTicksAndTickBitmap.success, 'Pool does not exist');
+
+    const [balance0, balance1, _stateWithoutTicksAndTickBitmap] = [
+      resBalance0.returnData,
+      resBalance1.returnData,
+      stateWithoutTicksAndTickBitmap.returnData,
+    ] as [bigint, bigint, DecodedStateMultiCallResultWithRelativeBitmaps];
+
+    const {
+      globalState: { tick },
+    } = _stateWithoutTicksAndTickBitmap;
+    const currentBitmapIndex = int16(
+      (BigInt(tick) / Constants.TICK_SPACING) >> 8n,
+    );
+
+    const buffer = this.getBitmapRangeToRequest();
+    const startBitMapIndex = currentBitmapIndex - buffer;
+    const endBitMapIndex = currentBitmapIndex + buffer;
+    const allBitMapIndices = Array.from(
+      { length: 2 * Number(buffer) + 1 },
+      () => startBitMapIndex + endBitMapIndex,
+    );
+
+    const ticksAndBitMaps = await Promise.all(
+      allBitMapIndices.map(relativeTick => {
+        return this.stateMultiContract.methods
+          .getAdditionalBitmapWithTicks(
+            this.factoryAddress,
+            this.token0,
+            this.token1,
+            relativeTick,
+            relativeTick,
+          )
+          .call() as [
+          TickBitMapMappingsWithBigNumber[],
+          TickInfoMappingsWithBigNumber[],
+        ];
+      }),
+    );
+
+    const _state: DecodedStateMultiCallResultWithRelativeBitmaps = {
+      ..._stateWithoutTicksAndTickBitmap,
+      tickBitmap: ticksAndBitMaps.flatMap(v => v[0]),
+      ticks: ticksAndBitMaps.flatMap(v => v[1]),
+    };
+
+    return [balance0, balance1, _state];
+  }
+
+  async _fetchInitStateMultiStrategies(
```

Comment:
If multicall is consistently failing for complex eth_call RPCs on zkEVM then shouldn't it always pick the multi step strategy for zkEVM?